.github/workflows/release.yaml

```diff
@@ -66,10 +75,39 @@ jobs:
         with:
           token: ${{ steps.generate_token.outputs.token }}
           fetch-depth: 0
-      - name: Tag Release
+      - name: Tag
         run: |
           git config --global user.password ${{ steps.generate_token.outputs.token }}
           git tag -d ${{env.VERSION}} 2> /dev/null || echo 'local ref does not exist'
           git push origin :${{env.VERSION}} 2> /dev/null || echo 'remote ref does not exist'
           git tag ${{env.VERSION}}
           git push origin ${{env.VERSION}}
+  release:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{env.APP_ID}}
+          private_key: ${{ secrets.TOKEN_EXCHANGE_GH_APP_PRIVATE_KEY }}
+          repository: ${{ github.repository }}
+          permissions: >-
+            {"contents": "write"}
```

Comment:
does this need `"pull_requests": "write"` permission, similar to [`buf`'s release workflow](https://github.com/bufbuild/buf/blob/7beb1e30eb91a52fb54f107f3a47515c5ad5168a/.github/workflows/release.yaml#L126)?

Reply:
this one does not, the one you reference is OP